yjit.c

```diff
@@ -622,8 +622,9 @@ rb_yjit_iseq_inspect(const rb_iseq_t *iseq)
     const char *path = RSTRING_PTR(rb_iseq_path(iseq));
     int lineno = iseq->body->location.code_location.beg_pos.lineno;
 
-    char *buf = ZALLOC_N(char, strlen(label) + strlen(path) + num_digits(lineno) + 3);
-    sprintf(buf, "%s@%s:%d", label, path, lineno);
+    const size_t size = strlen(label) + strlen(path) + num_digits(lineno) + 3;
```

Comment:
For this kind of pattern, as I understand it, it's normally recommended to call snprintf twice. For example, https://github.com/tekknolagi/fmt/blob/334b067ac0e5704a1da2b636d7028ca71a0d6eaa/fmt.h#L20

The first (with null buf) gives you the target size and then you can call it again with a buf of that size